pkgs/top-level/all-packages.nix

```diff
@@ -14627,11 +14627,14 @@ in
     ffmpeg = ffmpeg_2;
   };
 
-  spotify = callPackage ../applications/audio/spotify {
-    inherit (gnome2) GConf;
-    libgcrypt = libgcrypt_1_5;
-    libpng = libpng12;
-  };
+  spotify_mac = callPackage ../applications/audio/spotify/darwin.nix {};
```

Comment:
The thing is most of these have equivalent linux derivations that you should have the same name.